mlx/ops.cpp

```diff
@@ -2649,4 +2649,111 @@ array quantized_matmul(
   return out;
 }
 
+std::tuple<array, array, array> quantize(
```

Comment:
Does it make sense for this to be `vector<array>` rather than a tuple. (I'm assuming future  quantization methods will use this same entry point and possibly return more or fewer arrays).

Reply:
I don't know if that's the right thing TBH, just thinking out loud on how not to break the C++ API if we need to change or add to `quantize` in the future.

Reply:
Oh btw we didn't discuss this comment. I think tuple is fine for now. If anything it will force us to think about the change rather than implicitly changing what the return type is ie going from 3 return values to 2 is an API change even if a std::vector keeps the signature the same, right?